pkg/controllers/vdb/status_reconciler.go

```diff
@@ -213,8 +232,8 @@ func (s *StatusReconciler) calculateSubclusterStatus(ctx context.Context, sc *va
 		if pf.GetSubclusterOid() != "" {
 			curStat.Oid = pf.GetSubclusterOid()
 		}
-		if pf.GetSubclusterStatusType() != "" {
-			curStat.Type = pf.GetSubclusterStatusType()
+		if s.getSubclusterStatusType(podName) != "" {
+			curStat.Type = s.getSubclusterStatusType(podName)
```

Comment:
```suggestion
                scType := s.getSubclusterStatusType(podName)
		if scType != "" {
			curStat.Type = scType
```
Also, move this out of the loop. We only need to call the function once, not once for each pod in the subcluster.

Reply:
Good catch. Thanks for the suggestion.